lib/endpoint-proxy.js

```diff
@@ -521,11 +768,70 @@ class EndpointProxy {
     }
   }
 }
+class ServiceCaller {
+  constructor(client, serviceDeclaration, endpointDeclarations) {
+    if (client === void 0 || client === null) {
+      throw new Error('no database client provided');
+    }
+    if (serviceDeclaration === void 0 || serviceDeclaration === null) {
+      throw new Error('no service declaration provided');
+    }
+    const db = client.getConnectionParams().database;
+    if (db !== void 0 && db !== null) {
+      throw new Error('connection for client cannot specify database');
+    }
+    this._client = client;
```

Comment:
Because the $mlProxy field keeps a reference to the client, I believe it's not necessary for this class to keep a client field.